services/workflows-service/src/transaction/transaction.service.ts

```diff
@@ -14,86 +14,37 @@ export class TransactionService {
   ) {}
 
-  async create(payload: TransactionCreateDto): Promise<TransactionRecord> {
-    const transactionEntity: Prisma.TransactionRecordCreateInput = {
-      ...TransactionEntityMapper.toEntity(payload),
-      project: {
-        connect: { id: payload.projectId },
-      },
-      // #TODO: fix types - this is a workaround, prisma defines jsonb's as prsima "json input"
-      tags: payload.tags as any,
-      auditTrail: payload.auditTrail as any,
-      unusualActivityFlags: payload.unusualActivityFlags as any,
-      additionalInfo: payload.additionalInfo as any,
-      transactionCorrelationId: '',
-      transactionDate: '',
-      transactionAmount: 0,
-      transactionCurrency: '',
-    };
+  async createBulk({
+    transactionsPayload,
+    projectId,
+  }: {
+    transactionsPayload: TransactionCreateDto[];
+    projectId: TProjectId;
+  }) {
+    const response: Array<TransactionCreatedDto | { error: Error; correlationId: string }> = [];
 
-    return this.repository.create({ data: transactionEntity });
-  }
-
-  async createBatch(payload: TransactionCreateDto[]): Promise<{
-    txCreationResponse: Array<{
-      txid: string;
-      status: 'success' | 'failed';
-      txCorrelationId: string;
-      errorMessage?: string;
-    }>;
-    overallStatus: 'success' | 'partial';
-  }> {
-    // TEMP IMPLEMENTATION - REMOVE WHEN TASK BASED BATCH CREATE IS IMPLEMENTED
-
-    const txCreationResponse: Array<{
-      txid: string;
-      status: 'success' | 'failed';
-      txCorrelationId: string;
-      errorMessage?: string;
-    }> = [];
-    let overallStatus: 'success' | 'partial' = 'success';
-
-    for (const transaction of payload) {
-      const transactionEntity: Prisma.TransactionRecordCreateInput = {
-        ...TransactionEntityMapper.toEntity(transaction),
-        project: {
-          connect: { id: transaction.projectId },
-        },
-        // #TODO: fix types - this is a workaround, prisma defines jsonb's as prsima "json input"
-        tags: transaction.tags as any,
-        auditTrail: transaction.auditTrail as any,
-        unusualActivityFlags: transaction.unusualActivityFlags as any,
-        additionalInfo: transaction.additionalInfo as any,
-        transactionCorrelationId: '',
-        transactionDate: '',
-        transactionAmount: 0,
-        transactionCurrency: '',
-      };
-
-      await sleep(200);
-
-      let res;
+    for (const transactionPayload of transactionsPayload) {
       try {
-        res = await this.repository.create({ data: transactionEntity });
-        txCreationResponse.push({
-          txid: res.id,
-          txCorrelationId: transaction.correlationId,
-          status: 'success',
+        const transaction = await this.repository.create({
+          data: TransactionEntityMapper.toCreateData({
+            dto: transactionPayload,
+            projectId,
+          }),
         });
-      } catch (error) {
-        overallStatus = 'partial';
```

Comment:
this is exactly part of our bulk response you've just removed 
overallStatus: success|failed|partial

Reply:
Why is it important?
We return 201 if it was successful, and 209 for partial. I can return 500 if all of the all of the insertions were failed.